pkg/api/frontend_logging_test.go

```diff
@@ -0,0 +1,143 @@
+package api
+
+import (
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/getsentry/sentry-go"
+	glog "github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/models"
+	log "github.com/inconshreveable/log15"
+	. "github.com/smartystreets/goconvey/convey"
+)
+
+type logScenarioFunc func(c *scenarioContext, logs []*log.Record)
+
+func logSentryEventScenario(desc string, event FrontendSentryEvent, fn logScenarioFunc) {
+	Convey(desc, func() {
+		logs := []*log.Record{}
+		frontendLogger.SetHandler(log.FuncHandler(func(r *log.Record) error {
+			logs = append(logs, r)
+			return nil
+		}))
```

Comment:
Please add a `t.Cleanup()` function to reset the handler after the test is done, so tests don't leak state between each other.

Reply:
- [ ] add `t.Cleanup`